jenkins/packer/packer-build.jenkinsfile

```diff
@@ -0,0 +1,71 @@
+pipeline {
+    options {
+        timeout(time: 4, unit: 'HOURS')
+    }
+    agent {
+        docker {
+            label 'Jenkins-Agent-Ubuntu2004-X64-M52xlarge-Docker-Builder'
+            image 'opensearchstaging/ci-runner:ubuntu2004-x64-docker-buildx0.6.3-qemu5.0-awscli1.22-jdk11-v2'
+            args '-u root -v /var/run/docker.sock:/var/run/docker.sock'
+            alwaysPull true
+        }
+    }
+    parameters {
+        string(
+            name: 'PACKER_AGENT_JSON_NAME',
+            description: 'The JSON file name of type of Jenkins agent this job is building. e.g. jenkins-agent-al2-arm64.json. Files exist in https://github.com/opensearch-project/opensearch-ci/tree/main/packer',
+            trim: true
+        )
+    }
+    stages {
+        stage('Parameters Check') {
+            steps {
+                script {
+                    if (PACKER_AGENT_JSON_NAME.isEmpty()) {
+                        currentBuild.result = 'ABORTED'
+                        error("PACKER_AGENT_JSON_NAME parameter cannot be empty.")
+                    }
+                    print("Passing parameter checks.")
+                }
+            }
+        }
+        stage('packer-build') {
+            environment {
+                PACKER_BUILD_GIT_REPOSITORY_REFERENCE = "main"
+                PACKER_BUILD_GIT_REPOSITORY = "https://github.com/opensearch-project/opensearch-ci"
+            }
+            steps {
+                script {
+                    print("Getting $PACKER_AGENT_JSON_NAME from $PACKER_BUILD_GIT_REPOSITORY repository with reference $PACKER_BUILD_GIT_REPOSITORY_REFERENCE")
+                    checkout([$class: 'GitSCM', branches: [[name: PACKER_BUILD_GIT_REPOSITORY_REFERENCE ]], userRemoteConfigs: [[url: PACKER_BUILD_GIT_REPOSITORY ]]])
+                    def packer_jenkins_agent_json = readJSON file: "./packer/$PACKER_AGENT_JSON_NAME"
+                    withCredentials([
+                        string(credentialsId: "jenkins-agent-vpc-id", variable: 'vpc_id'),
+                        string(credentialsId: "jenkins-agent-subnet-id", variable: 'subnet_id'),
+                        string(credentialsId: "jenkins-agent-sg-id", variable: 'sg_id')
+                    ]){
+                        packer_jenkins_agent_json["variables"]["build-vpc"] = vpc_id
+                        packer_jenkins_agent_json["variables"]["build-subnet"] = subnet_id
+                        packer_jenkins_agent_json["variables"]["build-secgrp"] = sg_id
+                    }
+                    writeJSON file: "./packer/$PACKER_AGENT_JSON_NAME", json: packer_jenkins_agent_json, pretty: 2
```

Comment:
Suggest using `$PACKER_AGENT_JSON_NAME_substitute.json` so the original file will not get override.